src/containers/articles/NewArticle.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import ArticlesForm from '../../components/articles/ArticlesForm';
+import { createArticle, editArticle } from '../../actions/articleActions';
+import imageUploader from '../../utils/imageUploader';
+
+
+class CreateUpdateArticle extends Component {
+  state = {
+    title: '',
+    description: '',
+    body: '',
+    tagList: [],
+    image: '',
+  }
+
+  componentDidMount() {
+    const { create, articleData } = this.props;
+    if (!create) {
+      this.setState({
+        title: articleData.title,
+        description: articleData.description,
+        body: articleData.body,
+        tagList: articleData.tagList,
+        image: articleData.image,
+      });
+    }
+  }
+
+  handleChange = (event, element = 'title') => {
+    switch (element) {
+      case 'editor':
+        this.setState({
+          body: event.editor.getData(),
+        });
+        break;
+      case 'tags':
+        this.setState({
+          tagList: event,
+        });
+        break;
+      case 'image':
+        this.setState({
+          image: event,
+        });
+        break;
+
+      case 'description':
+        this.setState({
+          description: event.target.value,
+        });
+        break;
+      default:
+        this.setState({
+          title: event.target.value,
+        });
+        break;
+    }
+  }
+
+  handleSubmit = (event, create, slug = '') => {
+    event.preventDefault();
+    const { createArticle, editArticle, history } = this.props; //  eslint-disable-line
+    const {
+      title, description, body, tagList, image,
+    } = { ...this.state };
+    let data = {};
+    if (image) {
+      data = {
+        title, description, body, tagList, image,
+      };
+    } else {
+      data = {
+        title, description, body, tagList,
+      };
+    }
+
+    if (create) {
+      createArticle(data, history);
+    } else {
+      editArticle(slug, data, history);
+    }
+  }
+
+  render() {
+    let displayForm = '';
+    const {
+      image, description, body, tagList, title,
+    } = this.state;
+    const { create, articleData } = this.props;
+    if (localStorage.isLoggedIn) {
```

Comment:
This block can be used directly in the return section coz this abstraction does not add any value.